inst/htmljs/animint.js

```diff
@@ -1744,6 +1751,55 @@ var animint = function (to_select, json_file) {
 	return Selectors[s_name].duration;
       })
     ;
+    
+    // selector widgets
+    for(s_name in response.selectors) {
+      // create a widget for the selector
+      var selector_widget = element.append("button")
+        .text("Show " + s_name + " selectors")
+        .attr("class", "selector_widget")
+        .attr("id", s_name)
+      ;
+      // create a table for the buttons to go in - initially hidden
+      var selector_table = element.append("table")
+        .style("display", "none")
+        .attr("class", "selector_table")
+        .attr("id", s_name)
+      ;
+      // create a row in the table for the buttons
+      var selector_tr = selector_table
+        .append("tr")
+        .attr("id", s_name);
+      // create buttons for each level of the selector
+      response.selectors[s_name].levels.forEach(function(element) {
+        selector_tr
+          .append("button")
+          .text(element)
+          .attr("id", element)
+          .on("click", function() {
+            update_selector(s_name, this.textContent);
+          });
+      });
+      // on click, show the different options for that selector
+      d3.selectAll(".selector_widget")
+        .on("click", function() {
+          if(this.textContent.includes("Show")) {
+            // show table
+            d3.selectAll(".selector_table").select("#" + this.id)
+              .style("display", "");
+            // update widget text
+            this.innerHTML = "Hide " + this.id + " selectors";
+          } else {
+            // hide table
+            d3.selectAll(".selector_table").select("#" + this.id)
+              .style("display", "none");
+            // update widget text
+            this.innerHTML = "Show " + this.id + " selectors";
+          }
+        })
+      ;
```

Comment:
When clicking the "Show Species selector" button, I want to update the `selector_table` so that it is no longer hidden.  But this approach isn't working.  I think that there's a problem with the `d3.selectAll(".selector_table").select("#" + this.id)` statement, but I'm not sure what the problem is.  If anybody has any ideas, please let me know.